packages/trace-viewer/src/ui/attachmentsTab.tsx

```diff
@@ -139,12 +148,16 @@ export const AttachmentsTab: React.FunctionComponent<{
     {attachments.size ? <div className='attachments-section'>Attachments</div> : undefined}
     {[...attachments.values()].map((a, i) => {
       return <div className='attachment-item' key={attachmentKey(a, i)}>
-        <ExpandableAttachment attachment={a} />
+        <ExpandableAttachment attachment={a} highlight={isActiveAttachment(a, selectedAction)} />
       </div>;
     })}
   </div>;
 };
 
+function isActiveAttachment(attachment: Attachment, activeAction: ActionTraceEventInContext | undefined): boolean {
+  return activeAction?.attachments?.some(a => a.name === attachment.name && a.path === attachment.path && a.sha1 === attachment.sha1) ?? false;
```

Comment:
Perhaps just `activeAction?.attachments?.includes(attachment)`?

Reply:
tried that, but it doesn't work because referential equality is lost in https://github.com/microsoft/playwright/blob/2e01154bb57f5cb34c805092e0119742c46b95d8/packages/trace-viewer/src/ui/attachmentsTab.tsx#L93